tests/release-build.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+# test name and purpose
+echo ''
+echo '                        ##### Release Build Test #####'
+echo ''
+echo 'The purpose of this test is to ensure that nodeos was built without debugging'
+echo 'symbols. Debugging symbols enable software engineers to inspect and control a'
+echo 'running program with a debugging tool, but they significantly slow down'
+echo 'performance-critical applications like nodeos. Anyone intending to build and'
+echo 'install nodeos from source should perform a "release build," which excludes'
+echo 'debugging symbols to generate faster and lighter binaries.'
+echo ''
+# check for xxd
+if ! $(xxd --version 2>/dev/null); then
+    echo 'ERROR: Test requires xxd, but xxd was not found in your PATH!'
+    echo ''
+    echo 'The xxd hex dump tool can be installed as part of the vim-common package on most operating systems.'
+    exit 1
+fi
+# find nodeos
+[[ -z "$EOSIO_ROOT" && $(git --version) ]] && export EOSIO_ROOT="$(git rev-parse --show-toplevel)"
+[[ -z "$EOSIO_ROOT" ]] && export EOSIO_ROOT="$(echo $(pwd)/ | grep -ioe '.*/eos/' -e '.*/eosio/' -e '.*/build/' | sed 's,/build/,/,')"
+if [[ ! -f "$EOSIO_ROOT/build/bin/nodeos" && ! -f "$EOSIO_ROOT/build/programs/nodeos/nodeos" ]]; then
+    echo 'ERROR: nodeos binary not found!'
+    echo ''
+    echo 'Looked in the following places:'
+    echo "$ ls -la \"$EOSIO_ROOT/build/bin\""
+    ls -la "$EOSIO_ROOT/build/bin"
+    echo "$ ls -la \"$EOSIO_ROOT/build/programs/nodeos\""
+    ls -la "$EOSIO_ROOT/build/programs/nodeos"
+    echo 'Release build test not run.'
+    exit 2
+fi
+[[ -f "$EOSIO_ROOT/build/bin/nodeos" ]] && cd "$EOSIO_ROOT/build/bin" || cd "$EOSIO_ROOT/build/programs/nodeos"
+# run nodeos to generate state files
+./nodeos --config-dir "$(pwd)/config" --data-dir "$(pwd)/data" 1>/dev/null 2>/dev/null &
```

Comment:
I would prefer to see nodeos run with the `config` and `data` directories stored within another directory (that is clearly marked as intended for this test) within the initial current working directory of this script (which should be the build directory when this script is run by `ctest`). I'm a little concerned with potential directory conflicts between the nodeos directories this script uses and ones that may already exist for a user running ctest on their local machine.